pinot-spi/src/main/java/org/apache/pinot/spi/config/table/FieldConfig.java

```diff
@@ -254,6 +256,24 @@ public Builder withTierOverwrites(JsonNode tierOverwrites) {
       return this;
     }
 
+    public Builder withIndex(String indexId, IndexConfig indexConfig) {
```

Comment:
Is this related?

Reply:
I think it is the same case than `orNull` int iterator. I'm going to remove it for now